engines/python/setup/djl_python/rolling_batch/rolling_batch.py

```diff
@@ -64,29 +100,45 @@ def __init__(self, id: int, input_text: str, parameters: dict):
         self.id = id
         self.input_text = input_text
         self.parameters = parameters
-        self.next_token = None
+        self.next_token_str = None
         self.first_token = True
         self.last_token = False
+        self.token_cache = None
+        if parameters.get("details", False):
+            self.token_cache = []
 
     def __repr__(self):
         return f"<Request id: {self.id} Input {self.input_text} Parameters {self.parameters} Finished {self.last_token}>"
 
     def set_next_token(self,
-                       next_token: str,
+                       next_token: Union[Token, str],
```

Comment:
This is a Union, not Token object?

Reply:
https://stackoverflow.com/questions/48709104/how-do-i-specify-multiple-types-for-a-parameter-using-type-hints